src/state/mod.rs

```diff
@@ -284,11 +278,17 @@ impl IndexerState {
     }
 
     pub fn best_ledger(&self) -> Option<&Ledger> {
-        if let Some(head) = self.best_chain.last() {
-            Some(head.get_ledger())
-        } else {
-            None
+        let mut res = None;
+        if let Some(head_state_hash) = self.root_branch.longest_chain().first() {
+            // find the corresponding leaf ledger
+            for leaf in self.root_branch.leaves.values() {
+                if &leaf.block.state_hash == head_state_hash {
+                    res = Some(leaf.get_ledger());
+                    break;
+                }
+            }
         }
+        res
```

Comment:
Instead of break why not return the `Some(v)` there and `None` at the end. `let mut res` isn't necessary

Reply:
Late night brain! Will correct